unlock_indicator.c

```diff
@@ -1064,14 +1052,64 @@ void draw_image(uint32_t *resolution, xcb_drawable_t drawable) {
     cairo_destroy(xcb_ctx);
 }
 
+void draw_image(uint32_t* resolution, cairo_t* xcb_ctx) {
```

Comment:
It'd be helpful if you could add some brief documentation describing the method :)